torchao/float8/config.py

```diff
@@ -112,9 +116,23 @@ class Float8LinearConfig:
     cast_config_weight: CastConfig = CastConfig()
     cast_config_grad_output: CastConfig = CastConfig()
 
+    #
+    # Optional per-tensor configuration for `input`, `weight`, `grad_output` to
+    # calculate `grad_weight`, `grad_input`, and `grad_weight` respectively.
+    # If not specified, then the configuration from `cast_config_input`,
+    # `cast_config_weight` and `cast_config_grad_output`, respectively, is reused.
+    # TODO(future PR): maybe rename `cast_config_input` to 
+    # `cast_config_input_for_output`, etc, to make the names consistent, 
+    # will be BC-breaking.
+    #
+    cast_config_input_for_grad_weight: Optional[CastConfig] = None
+    cast_config_weight_for_grad_input: Optional[CastConfig] = None
+    cast_config_grad_output_for_grad_weight: Optional[CastConfig] = None
```

Comment:
Do we want to have two separate configs of `cast_config_grad_output_for_grad_weight` and `cast_config_grad_output_for_grad_input`?

Or, if we are just using `self.cast_config_grad_output` as `cast_config_grad_output_for_grad_input`, maybe add this info in comment?